azurerm/resource_arm_windows_virtual_machine_scale_set.go

```diff
@@ -0,0 +1,1275 @@
+										Required:     true,
+										ValidateFunc: azure.ValidateKeyVaultChildId,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"single_placement_group": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"source_image_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"source_image_reference": computeSvc.VirtualMachineScaleSetSourceImageReferenceSchema(),
+
+			"tags": tags.Schema(),
+
+			"terraform_should_roll_instances_when_required": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"timezone": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.VirtualMachineTimeZone(),
+			},
+
+			"upgrade_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(compute.Manual),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(compute.Automatic),
+					string(compute.Manual),
+					string(compute.Rolling),
+				}, false),
+			},
+
+			"winrm_listener": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				// Whilst the SDK allows you to modify this, the API does not:
+				//   Code="PropertyChangeNotAllowed"
+				//   Message="Changing property 'windowsConfiguration.winRM.listeners' is not allowed."
+				//   Target="windowsConfiguration.winRM.listeners"
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"protocol": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.HTTP),
+								string(compute.HTTPS),
+							}, false),
+						},
+
+						"certificate_url": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: azure.ValidateKeyVaultChildId,
+						},
+					},
+				},
+			},
+
+			"zone_balance": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+
+			"zones": azure.SchemaZones(),
+
+			// Computed
+			"unique_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmWindowsVirtualMachineScaleSetCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Compute.VMScaleSetClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	if features.ShouldResourcesBeImported() {
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Error checking for existing Windows Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(resp.Response) {
```

Comment:
Should this not be 
```suggestion
		if existing.ID != nil && *existing.ID != "" {
```

Reply:
this matches the Linux VMSS resource, so I think this is fine?

Reply:
hmm, many of the other resources are the other way around 🤷‍♀